src/main/java/com/conveyal/taui/controllers/BundleController.java

```diff
@@ -121,38 +128,47 @@ public static Bundle create (Request req, Response res) {
         // process async
         ExecutorServices.heavy.execute(() -> {
             try {
-                TDoubleList lats = new TDoubleArrayList();
-                TDoubleList lons = new TDoubleArrayList();
                 Set<String> seenFeedIds = new HashSet<>();
 
+                Envelope bundleBounds = new Envelope();
+                bundle.serviceStart = LocalDate.MAX;
+                bundle.serviceEnd = LocalDate.MIN;
                 bundle.feeds = new ArrayList<>();
                 bundle.totalFeeds = localFiles.size();
 
-
                 for (File file : localFiles) {
                     FeedSource fs = ApiMain.registerFeedSource(feed -> String.format("%s_%s", feed.feedId, bundle._id), file);
                     if (seenFeedIds.contains(fs.feed.feedId)) {
                         throw new Exception("Duplicate Feed ID found when uploading bundle");
                     }
 
-                    bundle.feeds.add(new Bundle.FeedSummary(fs.feed, bundle._id));
-                    fs.feed.stops.values().forEach(s -> {
-                        lats.add(s.stop_lat);
-                        lons.add(s.stop_lon);
-                    });
+                    Bundle.FeedSummary feedSummary  = new Bundle.FeedSummary(fs.feed, bundle._id);
+                    bundle.feeds.add(feedSummary);
+
+                    for (Stop s : fs.feed.stops.values()) {
+                        bundleBounds.expandToInclude(s.stop_lon, s.stop_lat);
+                    }
+
+                    if (bundle.serviceStart.isAfter(feedSummary.serviceStart)) {
+                        bundle.serviceStart = feedSummary.serviceStart;
+                    }
+
+                    if (bundle.serviceEnd.isBefore(feedSummary.serviceEnd)) {
+                        bundle.serviceEnd = feedSummary.serviceEnd;
+                    }
 
                     bundle.feedsComplete += 1;
                     Persistence.bundles.put(bundle);
                 }
 
-                // find the median stop location
-                // use a median because it is robust to outliers
-                lats.sort();
-                lons.sort();
-                // not a true median as we don't handle the case when there is an even number of stops
-                // and we are supposed to average the two middle value, but close enough
-                bundle.centerLat = lats.get(lats.size() / 2);
-                bundle.centerLon = lons.get(lons.size() / 2);
+                // TODO Handle crossing the antimeridian
```

Comment:
Realize this is basically a global TODO

Reply:
But not super important... based on https://en.wikipedia.org/wiki/180th_meridian I think we can put this off until we're doing work in three small islands in Fiji or Chukotka Autonomous Okrug, which is "mostly roadless and air travel is the main mode of passenger transport." I'd say it's only important that Analysis doesn't crash in these cases.